sway-lsp/src/handlers/notification.rs

```diff
@@ -0,0 +1,102 @@
+//! This module is responsible for implementing handlers for Language Server
+//! Protocol. This module specifically handles notifications.
+
+use crate::{capabilities, core::sync, global_state::GlobalState};
+use forc_pkg::PackageManifestFile;
+use lsp_types::{
+    DidChangeTextDocumentParams, DidChangeWatchedFilesParams, DidOpenTextDocumentParams,
+    DidSaveTextDocumentParams, FileChangeType,
+};
+
+pub(crate) async fn handle_did_open_text_document(
+    state: &GlobalState,
```

Comment:
My preference is to have these functions encapsulated in a struct that has `&GlobalState` as a member, so that we don't need to pass it into every function.